amethyst_assets/src/store/dir.rs

```diff
@@ -33,34 +34,66 @@ impl Directory {
 impl Store for Directory {
     type Error = IoError;
 
-    fn modified(&self, category: &str, id: &str, ext: &str) -> Result<u64, IoError> {
-        use std::fs::metadata;
-
+    fn modified(&self, category: &str, id: &str, exts: &[&str]) -> Result<u64, IoError> {
         let mut path = self.loc.clone();
 
         path.push(category);
         path.push(id);
-        path.set_extension(ext);
 
-        Ok(metadata(&path)?.modified()?.duration_since(UNIX_EPOCH).unwrap().as_secs())
+        for ext in exts {
+            match fs::metadata(path.with_extension(ext)) {
+                Ok(meta) => {
+                    return Ok(
+                        meta.modified()?
+                            .duration_since(UNIX_EPOCH)
+                            .unwrap()
+                            .as_secs(),
+                    );
+                }
+                Err(_) => (),
+            }
+        }
+
+        Err(IoError::new(
+            IoErrorKind::NotFound,
+            format!(
+                "no file {:?} with any of extensions {:?} found",
+                path,
+                exts
+            ),
+        ))
     }
 
     fn store_id(&self) -> StoreId {
         self.id
     }
 
-    fn load(&self, category: &str, name: &str, ext: &str) -> Result<Vec<u8>, IoError> {
+    fn load(&self, category: &str, name: &str, exts: &[&str]) -> Result<Vec<u8>, IoError> {
         use std::io::Read;
 
         let mut path = self.loc.clone();
 
         path.push(category);
         path.push(name);
-        path.set_extension(ext);
 
-        let mut v = Vec::new();
-        File::open(&path)?.read_to_end(&mut v)?;
+        for ext in exts {
+            match File::open(&path.with_extension(ext)) {
+                Ok(mut file) => {
+                    let mut v = Vec::new();
+                    file.read_to_end(&mut v)?;
+                    return Ok(v);
+                }
+                Err(_) => (),
```

Comment:
Yes that is intended. However, I will change to check for the ErrorKind and return the error if appropriated.